ofdb-core/src/usecases/review_tokens.rs

```diff
@@ -0,0 +1,18 @@
+use super::prelude::*;
+
+pub fn consume_review_token<R: ReviewTokenRepo>(
+    repo: &R,
+    review_nonce: &ReviewNonce,
+) -> Result<ReviewToken> {
+    let token = repo.consume_review_token(review_nonce)?;
+    debug_assert_eq!(review_nonce, &token.review_nonce);
+    if token.expires_at < Timestamp::now() {
+        return Err(Error::TokenExpired);
+    }
+    Ok(token)
+}
+
+pub fn delete_expired_review_tokens<R: ReviewTokenRepo>(repo: &R) -> Result<usize> {
+    let expired_before = Timestamp::now();
```

Comment:
Obtaining the current time is an impure side-effect and complicates testing. Passing those parameters all the way down from the API boundary would be preferable.